torch/testing/_internal/compilable_quantized_tensor.py

```diff
@@ -0,0 +1,366 @@
+        return torch.Tensor._make_wrapper_subclass(cls, shape, **kwargs)  # type: ignore[attr-defined]
+
+    def __init__(
+        self,
+        layout_tensor: MyDTypeLayout,
+        shape: torch.Size,
+        dtype: Optional[torch.dtype] = None,
+    ):
+        self.layout_tensor = layout_tensor
+
+    """__tensor_flatten__ and __tensor_unflatten__ are used to desugar the tensor into native Tensors/attributes and
+    reconstruct the tensor subclass instance from the desugared tensor and attributes, these are required to define
+    a Tensor subclass for torch.compile support
+    """
+
+    def __tensor_flatten__(self):  # type: ignore[no-untyped-def]
+        """
+        Given the class, returns the fields of the class as two lists
+        The first one contains any tensor fields such as int_data and scale as keys to a dictionary
+        The second one contains all other non tensor type fields as values of a list
+        """
+        return ["layout_tensor"], [self.shape, self.dtype]
+
+    @classmethod
+    def __tensor_unflatten__(  # type: ignore[no-untyped-def]
+        cls, tensor_data_dict, tensor_attributes, outer_size, outer_stride
+    ):
+        """
+        Given the flattened data from above, returns a class instance
+        tensor_data_dict contains the tensor fields of the class as a dictionary
+        tensor_attributes contains all other non tensor type fields
+        """
+        layout_tensor = tensor_data_dict["layout_tensor"]
+        shape, dtype = tensor_attributes
+        return cls(
+            layout_tensor,
+            shape if outer_size is None else outer_size,
+            dtype=dtype,
+        )
+
+    """classmethod that converts from a floating point Tensor (fp32/fp16/bf16) to the current dtype
+    """
+
+    @classmethod
+    def from_float(
+        cls,
+        input_float: torch.Tensor,
+        layout_type: LayoutType = PlainLayoutType(),
+    ) -> torch.Tensor:
+        dtype = torch.int16
+        scale = input_float.abs().max() / 255
+        int_data = (input_float / scale).to(torch.int8)
+        layout_tensor = PlainMyDTypeLayout(int_data, scale, PlainLayoutType())
+        return cls(layout_tensor, input_float.shape)
+
+    """[Optional] We can overwrite layout property of the Tensor to represent different packing formats
+    """
+
+    @property
+    def layout_type(self) -> LayoutType:
+        return self.layout_tensor.get_layout_type()
+
+    def dequantize(self, output_dtype=None):  # type: ignore[no-untyped-def]
+        """We can define a dequantize method to convert the quantized tensor to a floating point tensor"""
+        if output_dtype is None:
+            output_dtype = torch.get_default_dtype()
+        int_data, scale = self.layout_tensor.get_plain()
+        return int_data.to(output_dtype) * scale
+
+    def __repr__(self):  # type: ignore[no-untyped-def]
+        return (
+            f"{self.__class__.__name__}(data={self.dequantize()}, shape={self.shape}, "
```

Comment:
nit: a more standard way we've been defining __repr__ is something like `return f"MySubclass(layout_tensor={repr(self.layout_tensor)}, scale={repr(self.scale)})"`

Reply:
yeah it's a choice of what we want to display, for quantized tensor typically the internal representation is not that meaningful for end users because they will just be packed tensors, that's why we typically just print the dequantized value that it's trying to represent